lib/rrule/rule.rb

```diff
@@ -70,7 +70,11 @@ def each(floor_date: nil)
     end
 
     def next
-      enumerator.next
+      enumerator.next unless has_ended?
```

Comment:
My gut is that although it's probably fine to add a `has_ended?` method, I'd be hesitant to change the behavior of `next` - this change would cause infinite loops for anybody using this gem and relying on catching `StopIteration` to know when to stop looping. @nerdrew thoughts?

Reply:
In all due respect, no function should be expected to throw an error. If I call `#next` I either would expect a value or nothing at all.

Reply:
That's not how ruby defines [#next](https://ruby-doc.org/core-2.7.4/Enumerator.html#method-i-next) though. `next` is expected to throw `StopIteration`.

Reply:
I like how the `RRule` can be treated like an Enumerator.

Reply:
I think we'd have to do something like this to implement this correctly

```ruby
def has_ended?
  enumerator.peek
  false
rescue StopIteration
  true
end
```

But, IMO we should encourage users to use `rrule_enumerator = rule.each` instead of using our delegated `#next` b/c `#each` will return a composable ruby `Enumerator`.